### PR TITLE
Add react-compiler eslint plugin

### PR DESCRIPTION
**Pull request by ivy-lli**

### CODE REVIEW

File: packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx [hunk 1]
```diff
@@ -21,14 +21,13 @@ export const InputFieldWithBrowser = ({
...
   const [open, setOpen] = useState(false);
   const inputRef = useRef<HTMLInputElement>(null);
-  const cmsQuickFixPopoverRef = useRef<HTMLDivElement>(null);
   const { isFocusWithin, focusWithinProps } = useOnFocus(value, onChange);
   const { handleTextSelection, showQuickFix, getSelectedText, selection } = useTextSelection(inputRef);
 
   return (
     <Dialog open={open} onOpenChange={setOpen}>
       <BasicField label={label} message={message} style={{ flex: '1' }} {...focusWithinProps} className='badge-field' tabIndex={0}>
-        {isFocusWithin || cmsQuickFixPopoverRef.current ? (
```
Comment (ivy-lli) [inline packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx:31]: fyi @ivy-edp, @ivy-fhe 

It is not good to access ref values in the render part (beaks a rule of react)
So I wrote this stuff a little bit different. Should still be working as before

**Review by ivy-edp — APPROVED** [review]
thanks 👍

File: packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx (hunk 1 above)
Comment (ivy-fhe) [reply packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx:31]: :+1: goot to know

**Review by ivy-fhe — APPROVED** [review]
:+1:

File: packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx (hunk 1 above)
Comment (ivy-edp) [reply packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx:31]: I guess I need to take a closer look at these rules again 😬 😅 
https://react.dev/reference/rules

Comment (ivy-lli) [reply packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx:31]: wasn't aware of that case too, but this eslint plugin should check stuff like that :)

**Pull request merged**